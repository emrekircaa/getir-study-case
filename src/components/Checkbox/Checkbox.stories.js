import { Checkbox } from ".";

const meta = {
  title: "Components/Checkbox",
  component: Checkbox,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
};

export default meta;

export const Checked = {
  args: {
    label: "Checkbox",
    isChecked: true,
    count: 1,
  },
};
