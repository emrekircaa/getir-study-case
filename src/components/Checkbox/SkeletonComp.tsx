import React from "react";
import ContentLoader, { IContentLoaderProps } from "react-content-loader";

const Skeleton: React.FC<IContentLoaderProps> = props => (
  <ContentLoader
    speed={2}
    width={400}
    height={150}
    viewBox="0 0 400 150"
    backgroundColor="#f3f3f3"
    foregroundColor="#ecebeb"
    {...props}
  >
    <rect x="55" y="4" rx="5" ry="5" width="200" height="22" />
    <rect x="24" y="4" rx="5" ry="5" width="22" height="22" />
    <rect x="55" y="36" rx="5" ry="5" width="145" height="22" />
    <rect x="24" y="36" rx="5" ry="5" width="22" height="22" />
    <rect x="55" y="66" rx="5" ry="5" width="169" height="22" />
    <rect x="24" y="68" rx="5" ry="5" width="22" height="22" />
    <rect x="55" y="98" rx="5" ry="5" width="200" height="22" />
    <rect x="24" y="98" rx="5" ry="5" width="22" height="22" />
  </ContentLoader>
);

export default Skeleton;
