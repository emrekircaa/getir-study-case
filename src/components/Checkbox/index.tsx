import React from "react";
import styled from "styled-components";
import Tick from "../../assets/icons/Tick";
import { colors } from "../../theme";

interface CheckboxProps {
  onChange: () => void;
  isChecked: boolean;
  label: string;
  count: number;
}
export const Checkbox: React.FC<CheckboxProps> = ({
  onChange,
  isChecked,
  label,
  count,
}) => {
  return (
    <StyledCheckboxContainer
      data-testid="checkbox-container"
      onClick={() => onChange()}
    >
      <StyledCheckbox checked={isChecked}>
        <Tick color="#fff" />
      </StyledCheckbox>
      <StyledCheckboxLabel title={`${label} (${count})`}>
        <StyledCheckboxLabelText>{label}</StyledCheckboxLabelText>
        <StyledCheckboxCount> (1)</StyledCheckboxCount>
      </StyledCheckboxLabel>
    </StyledCheckboxContainer>
  );
};

const StyledCheckboxContainer = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 16.61px;
  &:hover {
    cursor: pointer;
  }
  &:last-child {
    margin-bottom: 0px;
  }
`;
const StyledCheckbox = styled.div<{ checked: boolean }>`
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${({ checked }) =>
    checked ? `${colors.gray500}` : `${colors.white}`};
  box-shadow: 0px 1px 7px rgba(93, 56, 192, 0.4);
  border-radius: 2px;
  height: 22px;
  width: 22px;
  transition: all 0.2s;
`;

const StyledCheckboxLabel = styled.div`
  margin-left: 8px;
  display: flex;
  width: 200px;
`;
const StyledCheckboxLabelText = styled.span`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  font-size: 14px;
  line-height: 18px;
  letter-spacing: 0.16px;
  color: ${colors.gray600};
`;

const StyledCheckboxCount = styled.span`
  font-size: 14px;
  line-height: 18px;
  letter-spacing: 0.16px;
  color: ${colors.gray600};
  margin-left: 4px;
`;
