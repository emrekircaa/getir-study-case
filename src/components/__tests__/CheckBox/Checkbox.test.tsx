import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Checkbox } from "../../Checkbox/index";

describe("Checkbox component", () => {
  it("renders with correct props", () => {
    const { getByTestId } = render(
      <Checkbox onChange={jest.fn()} isChecked={true} label="label" count={3} />
    );
    const checkbox = getByTestId("checkbox-container");
    expect(checkbox).toMatchSnapshot();
  });

  it("calls onChange when clicked", () => {
    const onChange = jest.fn();

    const { getByTestId } = render(
      <Checkbox onChange={onChange} isChecked={true} label="label" count={3} />
    );

    fireEvent.click(getByTestId("checkbox-container"));
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
