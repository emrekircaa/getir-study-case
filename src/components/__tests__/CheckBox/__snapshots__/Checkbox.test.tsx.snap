// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component renders with correct props 1`] = `
<div
  class="sc-beySPh ftvDKx"
  data-testid="checkbox-container"
>
  <div
    class="sc-guDLey gUihLF"
  >
    <svg
      color="#fff"
      fill="none"
      height="7"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 1 3.5 6 1 3.727"
        stroke="#fff"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.2"
      />
    </svg>
  </div>
  <div
    class="sc-dmyCSP dLnuKV"
    title="label (3)"
  >
    <span
      class="sc-hLQSwg tXaHH"
    >
      label
    </span>
    <span
      class="sc-eDLKkx fVYwBA"
    >
      (
      3
      )
    </span>
  </div>
</div>
`;
